### PR TITLE
feat: add schema validate admin API

### PR DESCRIPTION
**Pull request by kingluo**
### Description

https://lists.apache.org/thread/6lsbhh4c9ytjmrkr0jt66j5s4vyh482k

### Checklist

- [x] I have explained the need for this PR and the problem it solves
- [x] I have explained the changes or the new features added to this PR
- [x] I have added tests corresponding to this change
- [x] I have updated the documentation to reflect this change
- [x] I have verified that this change is backward compatible (If not, please discuss on the [APISIX mailing list](https://github.com/apache/apisix/tree/master#community) first)

### CODE REVIEW

**Review by moonming — COMMENTED** [review]
we need more test cases to cover invalid requests.

File: docs/en/latest/admin-api.md [hunk 1]
```diff
@@ -1514,3 +1514,52 @@ Proto resource request address: /apisix/admin/protos/{id}
...
+
+```bash
+curl http://127.0.0.1:9180/apisix/admin/schema/validate/routes \
+    -H 'X-API-KEY: edd1c9f034335f136f87ad84b625c8f1' -X POST -i -d '{
+    "uri": 1980,
+    "upstream": {
+        "scheme": "https",
+        "type": "roundrobin",
+        "nodes": {
+            "nghttp2.org": 1
```
[the remark this answers is not in the document]
Comment (kingluo) [reply docs/en/latest/admin-api.md:1549]: httpbin has been less stable in recent months, and now the community is using nghttp2 more.

**Review by Revolyssup — DISMISSED** [review]
LGTM

File: docs/en/latest/admin-api.md [hunk 2]
```diff
@@ -1514,3 +1514,52 @@ Proto resource request address: /apisix/admin/protos/{id}
 | content   | True    | String | content of `.proto` or `.pb` files | See [here](./plugins/grpc-transcode.md#enabling-the-plugin)         |
 | create_time | False    | Epoch timestamp (in seconds) of the created time. If missing, this field will be populated automatically.             | 1602883670                                       |
 | update_time | False    | Epoch timestamp (in seconds) of the updated time. If missing, this field will be populated automatically.             | 1602883670                                       |
+
+## Schema validate
```
Comment (shreemaan-abhishek) [inline docs/en/latest/admin-api.md:1518]:
```suggestion
## Schema validation
```

File: docs/en/latest/admin-api.md [hunk 3]
```diff
@@ -1514,3 +1514,52 @@ Proto resource request address: /apisix/admin/protos/{id}
...
+
+Check the validity of a configuration against its entity schema. This allows you to test your input before submitting a request to the entity endpoints of the Admin API.
+
+Note that this only performs the schema validation checks, checking that the input configuration is well-formed. Requests to the entity endpoint using the given configuration may still fail due to other reasons, such as invalid foreign key relationships or uniqueness check failures against the contents of the data store.
+
+### Schema validate API
```
Comment (shreemaan-abhishek) [inline docs/en/latest/admin-api.md:1524]:
```suggestion
### Schema validation
```

File: docs/en/latest/admin-api.md [hunk 4]
```diff
@@ -1514,3 +1514,52 @@ Proto resource request address: /apisix/admin/protos/{id}
...
+
+Schema validate request address: /apisix/admin/schema/validate/{resource}
```
Comment (shreemaan-abhishek) [inline docs/en/latest/admin-api.md:1526]:
```suggestion
Schema validation request address: /apisix/admin/schema/validate/{resource}
```

**Review by shreemaan-abhishek — CHANGES_REQUESTED** [review]

File: docs/en/latest/admin-api.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tao12345666333) [reply docs/en/latest/admin-api.md:1549]: yes. 

FYI We can use httpbun.org also

**Comment by kayx23** [comment]
Just a question: why not doing the schema check automatically **before** users sending requests to **any** Admin API endpoints? The current approach, that is to ask users to send requests to a specific `/validate` endpoint, looks rather manual.

File: docs/en/latest/admin-api.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (moonming) [reply docs/en/latest/admin-api.md:1549]: how about `https://mock.api7.ai/`?

**Review by moonming — CHANGES_REQUESTED** [review]
we need more test cases not only in `routes`

**Comment by moonming** [comment]
> Just a question: why not doing the schema check automatically **before** users sending requests to **any** Admin API endpoints? The current approach, that is to ask users to send requests to a specific `/validate` endpoint, feels rather manual.

The user's request already run the schema check. 
The purpose of this API is only to check the schema of requests, no more no less.

**Review by monkeyDluffy6017 — APPROVED** [review]

**Review by shreemaan-abhishek — APPROVED** [review]

**Review by moonming — APPROVED** [review]

**Review by kayx23 — APPROVED** [review]

File: docs/en/latest/admin-api.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (tao12345666333) [reply docs/en/latest/admin-api.md:1549]: @moonming SGTM.

**Pull request merged**